viam_cartographer_test.go

```diff
@@ -142,110 +158,35 @@ func TestNew(t *testing.T) {
+		test.That(t, is, test.ShouldNotBeNil)
 
 		timestamp1, err := svc.GetLatestMapInfo(context.Background())
 		test.That(t, err, test.ShouldBeNil)
-		_, err = svc.GetPointCloudMap(context.Background())
-		test.That(t, err, test.ShouldBeNil)
 		timestamp2, err := svc.GetLatestMapInfo(context.Background())
 		test.That(t, err, test.ShouldBeNil)
-
 		test.That(t, timestamp1.After(_zeroTime), test.ShouldBeTrue)
-		test.That(t, timestamp2.After(timestamp1), test.ShouldBeTrue)
+		test.That(t, timestamp1, test.ShouldResemble, timestamp2)
 
 		test.That(t, svc.Close(context.Background()), test.ShouldBeNil)
 	})
 
-	t.Run("Fails to create cartographer slam service with no sensor when feature flag enabled", func(t *testing.T) {
-		termFunc := testhelper.InitTestCL(t, logger)
-		defer termFunc()
-
-		dataDirectory, fsCleanupFunc := testhelper.InitInternalState(t)
-		defer fsCleanupFunc()
-
-		attrCfg := &vcConfig.Config{
-			Sensors:       []string{},
-			ConfigParams:  map[string]string{"mode": "2d"},
-			DataDirectory: dataDirectory,
-		}
-
-		svc, err := testhelper.CreateSLAMService(t, attrCfg, logger)
-		test.That(t, err, test.ShouldBeError, errors.New("error validating \"path\": \"sensors\" must not be empty"))
-		test.That(t, svc, test.ShouldBeNil)
-	})
-
-	t.Run("Failed creation of cartographer slam service with more than one sensor when feature flag enabled", func(t *testing.T) {
-		termFunc := testhelper.InitTestCL(t, logger)
-		defer termFunc()
-
-		dataDirectory, fsCleanupFunc := testhelper.InitInternalState(t)
-		defer fsCleanupFunc()
-
-		attrCfg := &vcConfig.Config{
-			Sensors:       []string{"lidar", "one-too-many"},
-			ConfigParams:  map[string]string{"mode": "2d"},
-			DataDirectory: dataDirectory,
-		}
-
-		_, err := testhelper.CreateSLAMService(t, attrCfg, logger)
-		test.That(t, err, test.ShouldBeError,
-			errors.New("configuring lidar camera error: 'sensors' must contain only one "+
-				"lidar camera, but is 'sensors: [lidar, one-too-many]'"))
-	})
-
-	t.Run("Failed creation of cartographer slam service with non-existing sensor when feature flag enabled", func(t *testing.T) {
-		termFunc := testhelper.InitTestCL(t, logger)
-		defer termFunc()
-
-		dataDirectory, fsCleanupFunc := testhelper.InitInternalState(t)
-		defer fsCleanupFunc()
-
-		attrCfg := &vcConfig.Config{
-			Sensors:       []string{"gibberish"},
-			ConfigParams:  map[string]string{"mode": "2d"},
-			DataDirectory: dataDirectory,
-			DataRateMsec:  testDataRateMsec,
-		}
-
-		_, err := testhelper.CreateSLAMService(t, attrCfg, logger)
-		test.That(t, err, test.ShouldBeError,
-			errors.New("error getting lidar camera "+
-				"gibberish for slam service: \"rdk:component:camera/gibberish\" missing from dependencies"))
-	})
-
-	t.Run("Successful creation of cartographer slam service with good lidar when feature flag enabled", func(t *testing.T) {
+	t.Run("Successful creation of cartographer slam service in non localization mode", func(t *testing.T) {
```

Comment:
Thie merges `Successful creation of cartographer slam service in non localization mode` and `Successful creation of cartographer slam service in non localization mode when feature flag enabled`